doc/concepts/interconnectivity.rst

```diff
@@ -74,6 +74,142 @@ full recipe.
    connectivity where the temptation to store all connections is even larger and
    each call to ``update`` will re-evaluate the corresponding callbacks.
 
+Cross-Simulator Interaction
+===========================
+
+The usual recipe can be used to declare connections to the world outside of
+Arbor similar to how internal (=both source and target are Arbor's
+responsibility) connections are handled.
+
+.. code-block:: c++
+
+    struct recipe(arb::recipe) {
+      // Rest as ever before
+      std::vector<arb::ext_cell_connection> external_connections_on(arb::cell_gid_type) const override {
+          return {{arb::cell_remote_label_type{42,  // External GID
+                                               23}, // per-gid tag
+                   arb::cell_local_label_type{"tgt"},
+                   weight,
+                   delay}};
+      }
+    };
+
+similarly
+
+.. code-block:: python
+
+    class recipe(arb.recipe):
+        def external_connections_on(self, gid):
+
+            return [arb.connection((42,      # external GID
+                                    32),     # tag
+                                    "tgt",
+                                    weight,
+                                    delay)]]
+
+Note that Arbor now recognizes two sets of ``GID``: An external and an internal
+set. This allows both Arbor and the coupled simulation to keep their own
+numbering schemes. However, internally Arbor will tag external cells and spikes
+by setting their ``GID``s' most significant bit. This _halves_ the effecively
+available ``GID``s.
```

Comment:
not clear to me what this means in practice - if this imposes a restrictions on the number of cells/maximum gid (internal or external), then we probably should provide a factory to generate the labels which does a check on the bounds.

Reply:
Yes, a model building error will be thrown if such a connection is found.